plugins/snapshots/erofs/erofs.go

```diff
@@ -445,6 +495,46 @@ func (s *snapshotter) commitBlock(ctx context.Context, layerBlob string, id stri
 	return nil
 }
 
+// generate a metadata-only EROFS fsmeta.erofs if all EROFS layer blobs are valid
+func (s *snapshotter) generateFsMeta(ctx context.Context, snapIDs []string) {
+	var blobs []string
+
+	if s.fsMergeThreshold == 0 || uint(len(snapIDs)) <= s.fsMergeThreshold {
+		return
+	}
+
+	t1 := time.Now()
+	mergedMeta := s.fsMetaPath(snapIDs[0])
+	// If the empty placeholder cannot be created (mainly due to os.IsExist), just return
+	if _, err := os.OpenFile(mergedMeta, os.O_CREATE|os.O_EXCL, 0644); err != nil {
+		return
+	}
+
+	for i := len(snapIDs) - 1; i >= 0; i-- {
+		blob := s.layerBlobPath(snapIDs[i])
+		if _, err := os.Stat(blob); err != nil {
+			return
+		}
+		blobs = append(blobs, blob)
+	}
+	args := append([]string{"--aufs", "--ovlfs-strip=1", "--quiet", mergedMeta + ".tmp"}, blobs...)
+	log.G(ctx).Infof("merging layers with mkfs.erofs %v", args)
+	cmd := exec.CommandContext(ctx, "mkfs.erofs", args...)
+	_, err := cmd.CombinedOutput()
+	if err != nil {
+		log.G(ctx).Infof("failed to generate merged fsmeta for %v", snapIDs[0])
```

Comment:
NIT: could it be a good idea to display what went wrong here ?

Reply:
yeah, I'm attending a conference now, will address later.